### PR TITLE
metamodel updates

### PR DESCRIPTION
**Pull request by prasad-okahu**
Added model name to type and updated docs

## Proposed changes
- Update metamodel docs
- Removed the entity instances eg types of inference services, vectorDBs etc from the core entity definition and created them as separate list of types.
- The entity type is now entity[.category.]type, eg inference.openai or model.llm.gpt-35-turbo

## Types of changes

What types of changes does your code introduce to this project?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have read the [CONTRIBUTING](https://github.com/monocle2ai/monocle/blob/main/CONTRIBUTING.md) doc
- [ ] I have signed the CLA
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: src/monocle_apptrace/metamodel/entities/README.md [hunk 1]
```diff
@@ -1,33 +1,44 @@
...
-| type | Monocle Entity type | True |
+| type | entity.category.type | True |
 
 ### MonocleEntity.Workflow
-Workflow ie the core application code. Supported types are -
+Workflow ie the core application code. 
+#### Workflow types
 - generic
 - langchain
 - llama_index
```
Comment (kshitiz-okahu) [inline src/monocle_apptrace/metamodel/entities/README.md:16]: llamaindex is a single word as seen on their official page
https://www.llamaindex.ai/

File: src/monocle_apptrace/metamodel/entities/entities.json [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        {
+            "name": "model",
+            "attributes" : [
+                {
+                    "attribute_name": "model_name",
+                    "attribute_description": "Model name",
+                    "required": true
+                }
+            ],
+            "categorie": [
```
Comment (oi-raanne) [inline src/monocle_apptrace/metamodel/entities/entities.json:18]: typo error for ```categories```

**Review by oi-raanne — CHANGES_REQUESTED** [review]

**Comment by prasad-okahu** [comment]
Updates per review feedback

File: src/monocle_apptrace/metamodel/entities/workflow_types.json [hunk 3]
```diff
@@ -0,0 +1,22 @@
+{
+    "entity" : "workflow",
+    "types": [
+        {
+            "name": "llama_index",
```
Comment (kshitiz-okahu) [inline src/monocle_apptrace/metamodel/entities/workflow_types.json:5]: spelling: llamaindex

**Review by kshitiz-okahu — APPROVED** [review]

File: src/monocle_apptrace/metamodel/entities/entities.json (hunk 2 above)
Comment (prasad-okahu) [reply src/monocle_apptrace/metamodel/entities/entities.json]: Updated per feedback

**Review by oi-raanne — APPROVED** [review]

**Pull request merged**